### PR TITLE
Include polkadot version in artifact path

### LINKED ISSUE #695: PVF: include polkadot's version into the artifact path
# ISSUE

## Overview

Compilation artifact filenames should contain the polkadot version. This way if there is a node upgrade, an old version can't override an artifact for the new node.

### PR DESCRIPTION
**Pull request by eagr**
closes #695

Could potentially be helpful to preserving caches when applicable, as discussed in #685 

kusama address: FvpsvV1GQAAbwqX6oyRjemgdKV11QU5bXsMg9xsonD1FLGK

### CODE REVIEW

File: polkadot/node/core/pvf/src/artifacts.rs
Comment (mrcnski) [inline polkadot/node/core/pvf/src/artifacts.rs:80]: I'd keep like this. Maybe in the future we can also add the `wasmtime` version. So it will look like `wasmtime_vXXX_polkadot_vYYY_...`

**Review by mrcnski — APPROVED** [review]
Thanks so much! If you'd like a tip and you have a Polkadot or Kusama address, you can add it to the PR body.

Format: `{kusama|polkadot|localtest} address: <SS58 Address>`
Doc: https://github.com/paritytech/substrate-tip-bot#pull-request-body
Example: https://github.com/paritytech/polkadot-sdk/pull/1720

Let me know if you want ideas for further contributions!

**Comment by eagr** [comment]
What a pleasant surprise! I'll get an account right away.

Btw, I was trying to add some version check before the execution, but it feels better to put it in the cache purging instead (someday maybe). So any good next issue for me? @mrcnski

**Comment by mrcnski** [comment]
> What a pleasant surprise! I'll get an account right away.

Nice! I'll tip you on https://github.com/paritytech/polkadot-sdk/pull/1774 as it's bigger and more likely for a tip to pass the referendum.

> Btw, I was trying to add some version check before the execution, but it feels better to put it in the cache purging instead (someday maybe). So any good next issue for me? @mrcnski

Cool! Since you're already working on this area, do you want to go ahead and implement artifact persistence? We would need the following:

1. https://github.com/paritytech/polkadot-sdk/issues/685
2. https://github.com/paritytech/polkadot-sdk/issues/677#issuecomment-1691882092 (hash after preparation, but only check it on node restart and not before every execution.)

**Comment by eagr** [comment]
Is there anything left unsolved here? I'd like to start working on those issues. It'd be if this could be merged first as they are closely related.

**Review by s0me0ne-unkn0wn — APPROVED** [review]
Looks good, I'd appreciate a review from @bkchr as I remember there were discussions about moving around the node version constant... @mrcnski or did you agree already?

**Comment by mrcnski** [comment]
Yeah, PR looks good but I would like to wait for @bkchr approval as well.

**Review by bkchr — APPROVED** [review]
Not the perfect place for the version. However, it is better than using an env variable.

**Pull request merged**
